enterprise/reporting/reporting.go

```diff
@@ -564,6 +587,11 @@ func (r *HandleT) Report(metrics []*types.PUReportedMetric, txn *sql.Tx) {
 	for _, metric := range metrics {
 		workspaceID := r.getWorkspaceID(metric.ConnectionDetails.SourceID)
 		metric := *metric
+
+		if r.IsEventNameTrackingDisabledForSource(metric.ConnectionDetails.SourceID) {
+			metric = transformMetricForEventNameTrackingDisabled(metric)
+		}
+
```

Comment:
A few suggestions that are somewhat idiomatic and a bit subjective:

* I would use [slices.Contains](https://pkg.go.dev/golang.org/x/exp/slices#Contains), for consistency and readability.
* I would not create new functions since the logic they are encapsulating is so simple, one line each assuming `slices.Contains` is used.

```suggestion
		if slices.Contains(sourcesWithEventNameTrackingDisabled, metric.ConnectionDetails.SourceID) {
	metric.StatusDetail.EventName = metric.StatusDetail.EventType
        }
```